gaia-rs/tests/abci/main.rs

```diff
@@ -0,0 +1,131 @@
+use bytes::Bytes;
+use gaia_rs::abci_handler::GaiaABCIHandler;
+use gaia_rs::config::AppConfig;
+use gaia_rs::genesis::GenesisState;
+use gaia_rs::store_keys::GaiaParamsStoreKey;
+use gaia_rs::GaiaApplication;
+use gears::baseapp::genesis::Genesis;
+use gears::baseapp::options::NodeOptions;
+use gears::baseapp::BaseApp;
+use gears::config::Config;
+use gears::core::tx::raw::TxRaw;
+use gears::crypto::info::SigningInfo;
+use gears::crypto::keys::ReadAccAddress;
+use gears::store::database::MemDB;
+use gears::tendermint::mock::{InitState, MockNode};
+use gears::tendermint::types::chain_id::ChainId;
+use gears::tendermint::types::proto::consensus::ConsensusParams;
+use gears::tendermint::types::proto::validator::ValidatorUpdate;
+use gears::tendermint::types::proto::Protobuf;
+use gears::tendermint::types::time::Timestamp;
+use gears::types::address::AccAddress;
+use gears::types::auth::fee::Fee;
+use gears::types::base::send::SendCoins;
+use gears::types::tx::body::TxBody;
+use keyring::key::pair::KeyPair;
+use prost::Message;
+
+mod scenario_1;
+
+struct User {
+    key_pair: KeyPair,
+    account_number: u64,
+}
+
+impl User {
+    pub fn address(&self) -> AccAddress {
+        self.key_pair.get_address()
+    }
+}
+
+fn setup_mock_node() -> (
+    MockNode<BaseApp<MemDB, GaiaParamsStoreKey, GaiaABCIHandler, GaiaApplication>, GenesisState>,
+    User,
+) {
+    let db = MemDB::new();
+    let node_options = NodeOptions::default();
+    let config: Config<AppConfig> = Config::default();
+    let app: BaseApp<MemDB, GaiaParamsStoreKey, GaiaABCIHandler, GaiaApplication> = BaseApp::new(
+        db,
+        GaiaParamsStoreKey::BaseApp,
+        GaiaABCIHandler::new(config),
+        node_options,
+    );
+    let chain_id = ChainId::default();
+
+    let mnemonic = "race draft rival universe maid cheese steel logic crowd fork comic easy truth drift tomorrow eye buddy head time cash swing swift midnight borrow";
+    let mnemonic = bip32::Mnemonic::new(mnemonic, bip32::Language::English).unwrap();
+    let key_pair = KeyPair::from_mnemonic(&mnemonic);
```

Comment:
As I remember we have similar code in `keypair`. And if I'm correct, we have to reuse it (maybe update before)

Reply:
Yes [here](https://github.com/rumos-io/gears/blob/0045ac03ad7a5dce396164b3d7f16c5012d5919b/keyring/src/key/pair/mod.rs#L68)